### PR TITLE
fixed csharp examples

### PR DESCRIPTION
**Pull request by ildoc**
Congrats on taking the first step to contributing to the [Sample Programs in Every Language](https://therenegadecoder.com/code/sample-programs-in-every-language/) repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [ ] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [ ] I plan to write the article for this code snippet
- [ ] I named the pull request using `<Sample Program> in <Language>` format
- [ ] I created/updated the language README
  - [ ] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
- [ ] I fixed #your-issue-number-here

## Notes

Feel free to put whatever you want here.

I fixed missing namespaces and indentations in c# examples.
(also refactored reverse string example)

### CODE REVIEW

**Comment by jrg94** [comment]
Hey! I love the changes. I just have a couple questions. 

1. What is the purpose of namespacing?
2. Does the Reverse String refactor work for things like emojis and whatnot?

I ask the second question because I believe I originally tried to do the array reversal, but it doesn't handle special characters properly like emojis and whatnot.

**Comment by chrboe** [comment]
This is actually a bit of a tricky issue.

First of all, the data type `char` in C# is actually a UTF-16 rune, so most emojis should just work when reversing the array.

The problem arises when using emojis (or oth

**Comment by chrboe** [comment]
This is actually a bit of a tricky issue.

First of all, the data type `char` in C# is actually a UTF-16 rune, so most emojis should just work when reversing the array.

The problem arises when using emojis (or other Unicode characters for that matter) with modifier characters. Take, for example, this emoji: 👴🏻
The way this works is that there's a "base" character (the old man) and a skin color modifier (white), which are both separate UTF-8 (or UTF-16 in the case of C#) characters.

Alas, simply reversing the string as a character array messes up these multi-character sequences, and finding a solution that does this well [gets rather complicated pretty quickly](https://stackoverflow.com/a/228062). 

That being said, the version of the function that this PR is supposed to replace might well work, but in my opinion it's a little too complex and language specific for a sample program.

tl;dr personally I'd focus on the problem of reversing a "regular" string rather than parsing emojis.

**Comment by jrg94** [comment]
I totally agree with keeping it simple. If I had my way, I'd probably remove the 'reverse a string' challenge as it has a lot of unnecessary complexity. 

When I set out to make this repo, I liked the idea of highlighting some of the nuances between languages. For instance, Java/C# leverage 16-bit characters while Python leverages unicode characters. In the latter case, we can actually reverse arbitrary unicode characters regardless of how the platform renders them. I think that's pretty cool.

Where things get sticky are languages like C# that have native library support for unicode characters. The original code snippet here is demonstrating that solution while the new solution is performing a brute force reversal of the character arrays. 

Personally, I don't mind either way, but I get a lot of comments making the case for either side. Usually, I say don't worry about emojis if that makes the solution overly complex. For example, lower-level languages like C/C++ typically don't have unicode support. 

In the case of C#, I don't see too much additional complexity from using the native libraries (other than what's happening under the hood). The solution is about 4 lines longer than the brute force solution provided, so I'm not really bothered either way. 

That said, I can see some issues with the original solution. What happens if those libraries lose support? How about if those libraries change or grow? That becomes kind of a maintenance headache for this repo, so simpler is probably better. 

So, long story short, I like your idea to keep it simple. Is there any way you'd like to rewrite the [contribution document](https://github.com/jrg94/sample-programs/blob/master/CONTRIBUTING.md) section on string reversals to make the strict character reversal solution clear? 😃

**Comment by jrg94** [comment]
For now, I'll merge in this solution. I think you make a pretty solid case, and we'll just have to be more consistent in the future.
